File: fxgl/src/main/java/com/almasb/fxgl/gameplay/NotificationListener.java
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.gameplay;

/**
 * Marks a service that wants to listen for notification events.
 *
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
public interface NotificationListener {

    /**
     * Fired on notification event.
     *
     * @param event the notification event
     */
    void onNotificationEvent(NotificationEvent event);
}

File: fxgl-base/src/main/kotlin/com/almasb/fxgl/gameplay/notification/NotificationListener.kt
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.gameplay.notification

/**
 * Listener for notification events.
 *
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
interface NotificationListener {

    /**
     * Fired on notification event.
     *
     * @param event the notification event
     */
    fun onNotificationEvent(event: NotificationEvent)
}